core/domain/feature_gating_services_test.py

```diff
@@ -0,0 +1,217 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for feature_gating_services.py."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core.domain import feature_gating_services as feature_services
+from core.domain import platform_parameter_domain as param_domain
+from core.domain import platform_parameter_registry as registry
+from core.platform import models
+from core.tests import test_utils
+import utils
+
+memcache_services = models.Registry.import_memcache_services()
+
+
+class FeatureGatingServicesTest(test_utils.GenericTestBase):
+    """Test for the feature gating services."""
+
+    def setUp(self):
+        super(FeatureGatingServicesTest, self).setUp()
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.user_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        registry.Registry.parameter_registry.clear()
+        # Parameter names that might be used in following tests.
+        param_names = ['parameter_a', 'parameter_b']
+        memcache_keys = [
+            param_domain.PlatformParameter.get_memcache_key(name)
+            for name in param_names]
+        memcache_services.delete_multi(memcache_keys)
+
+        self.param_1 = registry.Registry.create_platform_parameter(
```

Comment:
I suggest naming these parameters based on their stage since that's the key difference between them, and it may help with test readability.